modules/services/window-managers/xmonad.nix

```diff
@@ -75,27 +75,57 @@ in {
           by Home Manager.
         '';
       };
+
+      libFiles = mkOption {
+        type = types.attrs;
+        default = { };
+        example = literalExample ''
+          {
+            "Tools.hs" = pkgs.writeText "Tools.hs" '''
+               module Tools
+               where
+
+               screenshot = "scrot"
+             ''';
+          }
+        '';
+        description = ''
+          Additional files that will be saved in
+          <filename>~/.xmonad/lib/</filename>. The keys are the file names
+          the value is a path to the contents of the files.
+        '';
+      };
     };
   };
 
-  config = mkIf cfg.enable (mkMerge [
+  config = let
+    onChange = ''
+      echo "Recompiling xmonad"
+      $DRY_RUN_CMD ${config.xsession.windowManager.command} --recompile
+
+      # Attempt to restart xmonad if X is running.
+      if [[ -v DISPLAY ]] ; then
+        echo "Restarting xmonad"
+        $DRY_RUN_CMD ${config.xsession.windowManager.command} --restart
+      fi
```

Comment:
I would gate the auto restarting behind an option that defaults to false if it can have any unwanted behavior (closing applications, crashing, etc.).